lnrpc/routerrpc/router_server.go

```diff
@@ -318,25 +318,24 @@ func (s *Server) SendToRoute(ctx context.Context,
 		return nil, err
 	}
 
-	preimage, err := s.cfg.Router.SendToRoute(hash, route)
+	attempt, err := s.cfg.Router.SendToRoute(hash, route)
+	if attempt != nil {
```

Comment:
Odd stanza here. Why not first `if err != nil { return err }`?

Reply:
The issue is that it is possible that an htlc attempt was started and closed (new db entry) caused by an error. In that case, both `attempt` and `error` are non-nil. This gives precedence to what is recorded in the db (the `attempt`). This is also the only info you'd get if you would look at the payment afterwards or via `TrackPayment`.

Reply:
Comments added to code